Cloud changelog - June 15th (#5682)
mattermost/docs: source/install/cloud-changelog.md

Latest Mattermost Cloud releases:

- [Release 2022-06-15](#release-2022-06-15)
- [Release 2022-05-26](#release-2022-05-26)
- [Release 2022-05-12](#release-2022-05-12)
- [Release 2022-04-28](#release-2022-04-28)
- [Release 2022-04-13](#release-2022-04-13)
- [Release 2022-03-30](#release-2022-03-30)

## Release 2022-06-15

### Highlights

#### Free Forever Mattermost Cloud Plan
 - Mattermost Cloud now supports a free forever plan, Cloud Starter, for unlimited users for unlimited time.

>  - Access unlimited channels, playbooks, and boards across the Mattermost platform, with unlimited voice calls and screen sharing in Direct Messages.

Reviewer: This sentence may read to some as if calls in v7.0 are only available via DMs, though calls is a Channels integration available across all public and private channels (I believe).

Reviewer 2: I guess it depends on what we are referring to. 

- Cloud free (starter) plan has Calls limited to DMs only (essentially forcing 1-1s). 
- Cloud professional/enterprise have a temporary limit of 8 participants per call just for the time being (initial beta phase) but are not limited to any specific channel type.
- Self-hosted (any version/plan) has no limits.

Author: @jasonblais can confirm if we're just referring to Starter.

Reviewer 3: This note is specific to Starter. There is a separate highlight section in the changelog for `Calls (Beta)` below.

>  - The free plan includes a few workspace limits:

Reviewer 3: ```suggestion
 - The free plan includes a few workspace limits. [Upgrade to Professional for unlimited access](https://mattermost.com/pricing/):
```

    - Maximum 1 team. Any additional teams created during a trial or on a paid plan will be archived.
    - 10GB file storage across the platform, with 100MB upload limit.
    - Unlimited installed apps or plugins, with maximum of 5 enabled at one time.

>     - Access to 10,000 most recent messages.

Reviewer 4: I'd mention that history will be preserved

Reviewer: I believe history is preserved across all 4 limits above as well, though that archived history can only be accessed if a free user upgrades to a paid subscription plan.

#### User Interface (UI)
 - Applied new designs for the Login screen.
 - Changed some tooltips to appear when focused instead of just on hover.
 - The legacy **Settings > Advanced Settings > Enable Post Formatting** setting is now deprecated in favor of the new message formatting toolbar.

#### Administration
 - For new Cloud Workspaces, the default value for **Collapsed Reply Threads** in **System Console > Experimental > Features** is now set to **Always On**. Does not impact existing workspaces where this value is already set.
 - Mattermost Cloud Professional plan now includes a 250GB file storage limit.

#### API Changes
 - To allow Admins to retrieve contents of posts whether they are deleted or not, ``include_deleted`` query parameter was introduced to ``GetPost`` endpoint.

#### Performance
 - Reduced the number of calls made to ``viewChannel`` API during regular usage.
 - Added pagination to the ``getPostThread`` API calls.

### Bug Fixes
 - Fixed an issue with uploading SVG files.
 - Fixed an issue where thread posts were not left-aligned in compact message display mode.
 - Fixed an error about a missing column for the Shared Channels experimental feature.

>  - Fixed an issue where the channel menu drop-down TAB skipped the option "Move to...".

Reviewer: Was the issue present when a user pressed the TAB keyboard key?

 - Fixed an issue where the bulk import failed due to reply ``CreateAt`` being greater than that of the parent post.
 - Fixed an undefined error when leaving a channel with the Unreads filter enabled.

>  - Fixed an issue where clicking on a quick emoji reaction opened the right-hand side.

Reviewer: ```suggestion
 - Fixed an issue where clicking on a quick emoji reaction opened the right-hand pane.
```

>  - Fixed an issue where the keyboard focus did not go back to the post textbox after hitting Ctrl+Shift+P twice.

Reviewer: ```suggestion
 - Fixed an issue where the keyboard focus did not go back to the post textbox after hitting CTRL+SHIFT+P twice.
```

Reviewer: Was this issue present on macOS as well with users pressing CMD+SHIFT+P twice?

Latest Mattermost Cloud releases:

- [Release 2022-06-15](#release-2022-06-15)
- [Release 2022-05-26](#release-2022-05-26)
- [Release 2022-05-12](#release-2022-05-12)
- [Release 2022-04-28](#release-2022-04-28)
- [Release 2022-04-13](#release-2022-04-13)
- [Release 2022-03-30](#release-2022-03-30)

## Release 2022-06-15

### Highlights

#### Free Forever Mattermost Cloud Plan
 - Mattermost Cloud now supports a free forever plan, Cloud Starter, for unlimited users for unlimited time.
 - Access unlimited channels, playbooks, and boards across the Mattermost platform, with unlimited voice calls and screen sharing in Direct Messages.
 - The free plan includes a few workspace limits. [Upgrade to Professional for unlimited access](https://mattermost.com/pricing/):

>     - Maximum 1 team. Any additional teams created during a trial or on a paid plan will be archived. History will be preserved, though that archived history can only be accessed if a free user upgrades to a paid subscription plan.

Reviewer 5: > Any additional teams created during a trial or on a paid plan will be archived.
@jasonblais what do you mean they will be archived on a paid plan? Do you mean if they downgrade?

>     - Maximum 1 team. Any additional teams created during a trial or on a paid plan will be archived. History will be preserved, though that archived history can only be accessed if a free user upgrades to a paid subscription plan.

Reviewer 3: ```suggestion
    - Maximum 1 team. Additional teams can be created during a trial or on a paid plan, which are archived if downgraded back to Starter. History will be preserved, though that archived history can only be accessed if a free user upgrades to a paid subscription plan.
```

Reviewer 3: Good question @marianunez, clarified above that archiving refers to when/if the workspace is downgraded to Starter.

    - 10GB file storage across the platform, with 100MB upload limit. History will be preserved.
    - Unlimited installed apps or plugins, with maximum of 5 enabled at one time. History will be preserved.
    - Access to 10,000 most recent messages. History will be preserved.
    - Access to 500 most recently updated cards.
    - Maximum 5 saved views per board.

#### Calls (Beta)
 - Native voice calling and screen sharing is now available. This is a Channels-specific integration.

#### Apps Bar (Beta)

>  - The channel header is now decluttered to make it more obvious how to access Calls, Playbooks, and Boards when viewing a channel. All channel header icons registered by plugins are moved to the new Apps Bar.

Reviewer 6: The channel header will only contain the Calls button, while Playbooks and Boards will be in the Apps Bar
![image](https://user-images.githubusercontent.com/3924815/172567596-d72a39ff-b027-4494-9e9c-5a365fee5456.png)

    - Access to 500 most recently updated cards.
    - Maximum 5 saved views per board.

#### Calls (Beta)
 - Native voice calling and screen sharing is now available. This is a Channels-specific integration.

#### Apps Bar (Beta)
 - The channel header is now decluttered to make it more obvious how to access Calls, Playbooks, and Boards when viewing a channel. All channel header icons registered by plugins are moved to the new Apps Bar, while Calls remains in the channel header.

#### Collapsed Reply Threads (General Availability)
 - [Collapsed Reply Threads](XXXXXX link to CRT GA blog post) is now generally available. Please reference [this article](link to KB article XXXXXX) for more information.

### Improvements

#### User Interface (UI)
 - Applied new designs for the Login screen.
 - Changed some tooltips to appear when focused instead of just on hover.
 - The legacy **Settings > Advanced Settings > Enable Post Formatting** and **Settings > Advanced Settings > Preview Pre-release Features** settings are now deprecated in favor of the new message formatting toolbar.
 - Romanian language support was downgraded to Beta.

#### Administration

>  - For new Cloud Workspaces, the default value for **Collapsed Reply Threads** in **System Console > Experimental > Features** is now set to **Always On**. Does not impact existing workspaces where this value is already set.

Reviewer: We're moving this config setting from Experimental to Site Configuration > Posts via https://github.com/mattermost/mattermost-webapp/pull/10465

 - Mattermost Cloud Professional plan now includes a 250GB file storage limit.

>  - Default password requirements have been loosened to 8 characters and no numeric, casing or special characters required by default. These requirements can be configured by the System Admin as needed in the **System Console**.

Reviewer: ```suggestion
 - Default password requirements have been loosened to eight characters and no numeric, casing, or special characters required by default. These requirements can be configured by the System Admin as needed in the **System Console**.
```

Reviewer: What's the path in the System Console to these configuration options?

Latest Mattermost Cloud releases:

- [Release 2022-06-15](#release-2022-06-15)
- [Release 2022-05-26](#release-2022-05-26)
- [Release 2022-05-12](#release-2022-05-12)
- [Release 2022-04-28](#release-2022-04-28)
- [Release 2022-04-13](#release-2022-04-13)
- [Release 2022-03-30](#release-2022-03-30)

## Release 2022-06-15

### Highlights

#### Workspace Limits

>  - Mattermost is releasing a free-forever Cloud offering.

Reviewer: called Cloud Starter?

>  - Mattermost is releasing a free-forever Cloud offering.

Reviewer 3: Per Maria's note below, propose something like

```
#### Free Forever Mattermost Cloud Plan
 - Mattermost Cloud now supports a free forever plan, Cloud Starter, for unlimited users for unlimited time.
 - Access unlimited channels, playbooks, and boards across the Mattermost platform, with unlimited voice calls and screen sharing in Direct Messages.
 - The free plan includes a few workspace limits:
    - Maximum 1 team. Any additional teams created during a trial or on a paid plan will be archived.
    - 10GB file storage across the platform, with 100MB upload limit.
    - Unlimited installed apps or plugins, with maximum of 5 enabled at one time.
    - Access to 10,000 most recent messages.
    - Access to 500 most recently updated cards.
    - Maximum 5 saved views per board.
```

>    - Added a new API endpoint ``GET /api/v4/usage/posts`` that returns the current usage limits for messages.

Reviewer: Do API updates like this get grouped under API Changes? There's a few API updates that may make sense there.

   - Team Limit changes now propagate automatically.
   - Added a new endpoint ``DELETE /api/v4/teams/{team_id:[A-Za-z0-9]+}/except`` which archives all teams except for the ``team_id`` passed to the endpoint.
   - A server configured to use the Cloud Freemium limits now guards against enabling more integrations than the configured limit.

> Users are restricted from creating teams when at or above the teams limit for their plan.

Reviewer: new bullet point?

   - Fixed API ``GET /api/v4/usage/posts`` to include user posts only.
   - Improved performance of API ``GET /api/v4/usage/posts`` by enabling caching of posts count.

>    - Users will be prevented from unarchiving teams if they're at or over the teams limit for their plan.

Reviewer 5: Most of these we can omit because they are backend implementation details not relevant for Cloud. @jasonblais wondering if we should just have a generic description here on Cloud Freemium being released and the workspace limits that should be known

#### Calls (Beta)

>  - Native voice calling and screen sharing is now available. This is a Channels-specific integration.

Reviewer: @itao - Any concerns with referring to Calls as "native voice calling"?

Reviewer 3: As a second note, do we need to mention `This is a Channels-specific integration.`? 0/5

Author: > do we need to mention This is a Channels-specific integration.?

In the v7.0 changelog @streamer45 suggested mentioning it and I added it here too (unless I misunderstood).

Reviewer 2: @itao Can confirm but the way we are going to market Calls currently is as a Channels integration and not a new product of the suite. That said, I don't have a strong opinion on whether we should stress this in release notes/changelogs.

#### Apps Bar (Beta)

>  - Channel header is now decluttered when a new configuration setting via **System Console > Experimental > Features > Enable App Bar** is enabled, the channel header will be decluttered to make it more obvious how to access Calls, Playbooks, and Boards when viewing a channel. All channel header icons registered by plugins will be moved to the new Apps Bar when the configuration setting is enabled. We recommend enabling the Apps Bar for servers with Calls enabled since the Apps Bar helps make space for the dedicated **Start Call** button in the channel header.

Reviewer: We need to align on whether it's an "Apps Bar" or an "App Bar". Docs is currently using "Apps Bar" to match v7.0 training via the PM organization.

#### Collapsed Reply Threads (General Availability)
 - [Collapsed Reply Threads](XXXXXX link to CRT GA blog post) is now generally available. Please reference [this article](link to KB article XXXXXX) for more information and guidance for enabling the feature.

### Improvements

#### User Interface (UI)

>  - Added a modal to allow users to switch subscription plans pre-trial.

Reviewer 3: Does "pre-trial" mean before they've started a trial or before the trial has expired?

Reviewer 5: I believe they are referring to before they have started a trial. This can probably either be omitted or add a general note about our new pricing page that outlines the different cloud plans with the ability to subscribe/change plans.

Reviewer 3: Got it, yea I think we can omit. I've now added a link to the pricing page to where we share the free forever Cloud plan

 - Applied new designs for the Login screen.
 - Changed some tooltips to appear when focused instead of just on hover.

>  - The legacy ``Enable post formatting`` setting is now hidden when the Message Formatting Toolbar is enabled.

Reviewer 7: ... hidden when the Message Formatting Toolbar is enabled > ... deprecated in favor of the new message formatting toolbar.

Reviewer 3: What is the "Enable post formatting" setting in the UI, and where is it accessible?

>  - Updated Apps Framework to version 1.1.0 to add improved logging.

Reviewer 3: Logging not available in Cloud, propose removing

#### Administration

>  - The default for ``ServiceSettings.CollapsedThreads`` has been changed to "always_on". This does not affect existing configurations where this value is already set to some other value.

Reviewer 3: ```suggestion
 - For new Cloud Workspaces, the default value for **Collapsed Reply Threads** in System Console > Experimental > Features is now set to **Always On**. Does not impact existing workspaces where this value is already set.
```

Reviewer 3: Cloud workspaces never use config settings, let's always refer to the System Console UI option instead.

>  - The default for ``ServiceSettings.CollapsedThreads`` has been changed to "always_on". This does not affect existing configurations where this value is already set to some other value.

Reviewer 3: Add "Mattermost Cloud Professional plan now includes a 250GB file storage limit."

>  - To enable or disable the App Bar, a new config setting ``ExperimentalSettings.EnableAppBar`` was added. 

Reviewer: App or Apps Bar?

>  - To enable or disable the App Bar, a new config setting ``ExperimentalSettings.EnableAppBar`` was added. 

Reviewer 3: Does not apply to Cloud (at least couldn't find such an option in the System Console)

>  - The value of ``ServiceSettings.TrustedProxyIPHeader`` defaults to empty from now on.

Reviewer 3: Does not apply to Cloud

#### API Changes
 - To allow Admins to retrieve contents of posts whether they are deleted or not, ``include_deleted`` query parameter was introduced to ``GetPost`` endpoint.

#### Performance
 - Reduced the number of calls made to ``viewChannel`` API during regular usage.
 - Added pagination to the ``getPostThread`` API calls.

### Bug Fixes
 - Fixed an issue with uploading SVG files.
 - Fixed an issue where thread posts were not left-aligned in compact message display mode.
 - Fixed an error about a missing column for the Shared Channels experimental feature.
 - Fixed an issue where the channel menu drop-down TAB skipped the option "Move to...".
 - Fixed an issue where the bulk import failed due to reply ``CreateAt`` being greater than that of the parent post.
 - Fixed an undefined error when leaving a channel with the Unreads filter enabled.

>  - Fixed an issue in X-Forwarded-For header parsing when no whitespaces between values are provided.

Reviewer 3: I'm not sure what's the user impact, so not sure if it applies to Cloud

 - The free plan includes a few workspace limits. [Upgrade to Professional for unlimited access](https://mattermost.com/pricing/):
    - Ability to start a free 30-day Enterprise trial, with access to all features during the trial period.
    - Maximum 1 team. Additional teams can be created during a trial or on a paid plan, which are archived if downgraded back to Starter. History will be preserved, though that archived history can only be accessed if a free user upgrades to a paid subscription plan.
    - 10GB file storage across the platform, with 100MB upload limit. History will be preserved.
    - Unlimited installed apps or plugins, with maximum of 5 enabled at one time. History will be preserved.
    - Access to 10,000 most recent messages. History will be preserved.
    - Access to 500 most recently updated cards.
    - Maximum 5 saved views per board.

#### Calls (Beta)
 - [Native voice calling and screen sharing](https://docs.mattermost.com/channels/make-calls.html) is now available. This is a Channels-specific integration.

#### Collapsed Reply Threads (General Availability)

>  - [Collapsed Reply Threads](https://docs.mattermost.com/channels/organize-conversations.html) is now generally available. Please reference [this article](https://support.mattermost.com/hc/en-us/articles/6880701948564) for more information and guidance for enabling the feature.

Reviewer 7: Collapsed Reply Threads](https://docs.mattermost.com/channels/organize-conversations.html) is now generally available and enabled by default for all Cloud users. Administrators may disable it in the System Console > Posts section.

#### Apps Bar (Beta)
 - The channel header is now decluttered to make it more obvious how to access Calls, Playbooks, and Boards when viewing a channel. All channel header icons registered by plugins are moved to the new Apps Bar, while Calls remains in the channel header.

### Improvements

#### User Interface (UI)
 - Applied new designs for the Login screen.
 - Changed some tooltips to appear when focused instead of just on hover.
 - The legacy **Settings > Advanced Settings > Enable Post Formatting** and **Settings > Advanced Settings > Preview Pre-release Features** settings are now deprecated in favor of the the new formatting toolbar.
 - Romanian language support was downgraded to Beta.

#### Administration

>  - For new Cloud Workspaces, the default value for **Collapsed Reply Threads** in **System Console > Posts** is now set to **Always On**. Does not impact existing workspaces where this value is already set.

Reviewer 7: For all Cloud Workspaces, the default value for **Collapsed Reply Threads** in **System Console > Posts** is now set to **Always On**. You may choose a different [configuration](https://docs.mattermost.com/configure/configuration-settings.html#collapsed-reply-threads) as desired.
